frontend/src/components/dashboard/noticeOfWorkHomePage/NoticeOfWorkHomePage.js

```diff
@@ -12,20 +16,99 @@ import { getNoticeOfWorkList, getNoticeOfWorkPageData } from "@/selectors/notice
 
 const propTypes = {
   fetchNoticeOfWorkApplications: PropTypes.func.isRequired,
-  // use NoW custom prop once this feature is fully implemented
-  // eslint-disable-next-line react/forbid-prop-types
-  noticeOfWorkApplications: PropTypes.arrayOf(PropTypes.any).isRequired,
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
+  location: PropTypes.shape({ search: PropTypes.string }).isRequired,
+  pageData: CustomPropTypes.pageData.isRequired,
+  noticeOfWorkApplications: PropTypes.arrayOf(CustomPropTypes.nowApplication).isRequired,
 };
 
 export class NoticeOfWorkHomePage extends Component {
-  state = { isLoaded: false };
+  params = queryString.parse(this.props.location.search);
+
+  state = {
+    isLoaded: false,
+    params: {
+      page: Strings.DEFAULT_PAGE,
+      per_page: Strings.DEFAULT_PER_PAGE,
+      ...this.params,
+    },
+  };
 
   componentDidMount() {
-    this.props.fetchNoticeOfWorkApplications().then(() => {
-      this.setState({ isLoaded: true });
-    });
+    const params = this.props.location.search;
+    const parsedParams = queryString.parse(params);
+    const { page = this.state.params.page, per_page = this.state.params.per_page } = parsedParams;
+    if (params) {
+      this.renderDataFromURL();
+    } else {
+      this.props.history.push(
+        router.NOTICE_OF_WORK_APPLICATIONS.dynamicRoute({
+          page,
+          per_page,
+        })
+      );
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged) {
+      this.renderDataFromURL(nextProps);
+    }
   }
 
+  componentWillUnmount() {
+    this.setState({ params: {} });
+  }
+
+  renderDataFromURL = (nextProps) => {
+    const params = nextProps ? nextProps.location.search : this.props.location.search;
+    const parsedParams = queryString.parse(params);
+    this.setState(
+      {
+        params: parsedParams,
+        isLoaded: false,
+      },
+      () =>
+        this.props.fetchNoticeOfWorkApplications(parsedParams).then(() => {
+          this.setState({ isLoaded: true });
+        })
+    );
+  };
+
+  handleSearch = (searchParams = {}, clear = false) => {
+    const persistedParams = clear ? {} : this.state.params;
+    const updatedParams = {
+      // Default per_page -- overwrite if provided
+      per_page: Strings.DEFAULT_PER_PAGE,
+      // Start from existing state
+      ...persistedParams,
+      // Overwrite prev params with any newly provided search params
+      ...searchParams,
+      // Reset page number
+      page: Strings.DEFAULT_PAGE,
+    };
+
+    this.props.history.push(router.NOTICE_OF_WORK_APPLICATIONS.dynamicRoute(updatedParams));
+    this.setState(
+      {
+        params: updatedParams,
+      },
+      // Fetch parties once state has been updated
+      () => this.renderDataFromURL()
+    );
+  };
```

Comment:
Same thing on Jon's PR: 

`this.props.history.push(router.NOTICE_OF_WORK_APPLICATIONS.dynamicRoute(updatedParams));`
Will trigger ComponentWillReceiveProps to detect a change in the location, which will invoke `renderdataFromUrl`, so the last return shouldnt be needed.

Reply:
Chatted IRL